### PR TITLE
Add module account_statement_operation_rule

### PR DESCRIPTION
**Pull request by guewen**
# account_statement_operation_rule

This module allows to automatically apply "operations" on the bank statements reconciliations according to rules.
Example of a rule: Apply the operation "Put the rest of the balance on account 4086" When the remaining balance is between -5 and 0.

More details: https://github.com/guewen/bank-statement-reconcile/blob/8.0-account_statement_operation_rule/account_statement_operation_rule/README.rst
# account_statement_operation_rule_dunning_fees

Extension of `account_statement_operation_rule`, adds a rule to handle the dunning fees.

More details: https://github.com/guewen/bank-statement-reconcile/blob/8.0-account_statement_operation_rule/account_statement_operation_rule_dunning_fees/README.rst

### CODE REVIEW

File: account_statement_operation_rule/i18n/fr.po [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+#. module: account_statement_operation_rule
+#: field:account.statement.operation.rule,create_date:0
+msgid "Created on"
+msgstr ""
+
+#. module: account_statement_operation_rule
+#: field:account.statement.operation.rule,currencies:0
+#: selection:account.statement.operation.rule,rule_type:0
+msgid "Currencies"
+msgstr "Devise"
```
Comment (mdietrichc2c) [inline account_statement_operation_rule/i18n/fr.po]: Use the plural "Devises"

File: account_statement_operation_rule/i18n/fr.po [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+#. module: account_statement_operation_rule
+#: field:account.statement.operation.rule,name:0
+msgid "Name"
+msgstr "Nom"
+
+#. module: account_statement_operation_rule
+#: field:account.statement.operation.rule,operations:0
+msgid "Operations"
+msgstr "Opération"
```
Comment (mdietrichc2c) [inline account_statement_operation_rule/i18n/fr.po]: Plural as well, "Opérations"

**Comment by mdietrichc2c** [comment]
2 translations errors, but the rest LGTM :+1:

File: .travis.yml [hunk 3]
```diff
@@ -15,6 +15,7 @@ install:
   - git clone https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
   - export PATH=${HOME}/maintainer-quality-tools/travis:${PATH}
   - travis_install_nightly
+  - git clone https://github.com/OCA/account-financial-tools $HOME/account-financial-tools -b ${VERSION}
```
Comment (pedrobaeza) [inline .travis.yml]: Add this to oca_dependencies.txt instead. Anyways, why is this needed?

File: account_statement_operation_rule/static/src/js/account_widgets.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                                      [self.st_line.id,
+                                       _.pluck(self.get("mv_lines_selected"), 'id')])
+                .then(function (operations) {
+                    _.each(operations, function(operation_id) {
+                        preset_btn = self.$("button.preset[data-presetid='" + operation_id + "']");
+                        preset_btn.click();
+                        self.addLineBeingEdited();
+                    });
+                });
+            // if (self.is_valid) {
```
Comment (pedrobaeza) [inline account_statement_operation_rule/static/src/js/account_widgets.js]: Remove this commented code

File: account_statement_operation_rule_dunning_fees/README.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License
+
+Bank Statement Operation Rules with Dunning Fees
```
Comment (pedrobaeza) [inline account_statement_operation_rule_dunning_fees/README.rst:4]: Isn't this called *_fees in general without the word dunning as it serves the same for any kind of fee?

Comment (guewen) [inline account_statement_operation_rule_dunning_fees/README.rst:4]: Because it depends on `account_credit_control_dunning_fees`

File: .travis.yml (hunk 3 above)
Comment (guewen) [inline .travis.yml]: Needed for the dependency on `account_credit_control_dunning_fees`. I guess the `oca_dependencies.txt` file was not existing at the time of creation of this PR. I'll change that.

**Comment by guewen** [comment]
Updated with the requested changes

**Comment by pedrobaeza** [comment]
Maybe you should squash your 50 commits in a few less, and more being a new module.

**Comment by guewen** [comment]
@pedrobaeza I reworked the history

File: .travis.yml [hunk 6]
```diff
@@ -31,6 +31,7 @@ install:
   - git clone --depth=1 https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
   - export PATH=${HOME}/maintainer-quality-tools/travis:${PATH}
   - travis_install_nightly
+  - git clone https://github.com/OCA/account-financial-tools $HOME/account-financial-tools -b ${VERSION}
```
Comment (pedrobaeza) [inline .travis.yml]: You have to remove this line because it's already declared in oca_dependencies.txt

File: account_statement_operation_rule/README.rst [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+This module complements the Reconciliation of the bank statements.  When
+the bank statement matches one or more journal entry for a line and
+there is a remaining balance, Odoo proposes you to click on buttons that
+will generate write-off entries according to pre-configured *Statement
+Operation Templates*. The aim of this module is to automatically click
+for you on these buttons (i.e. create the write-off journal entries)
+when some rules are respected, rules that you can configure.
+
+It contains 2 types of rule (but can be extended with additional rules),
```
Comment (pedrobaeza) [inline account_statement_operation_rule/README.rst]: s/rule/rules

File: account_statement_operation_rule/README.rst [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+described below:
+
+Roundings
```
Comment (pedrobaeza) [inline account_statement_operation_rule/README.rst:18]: Use `*` for making points

File: account_statement_operation_rule/README.rst [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+Usage
+-----
+
+When you use the *Reconcile* button of a bank statement, Odoo
+automatically proposes you matching journal entries for each statement
+line.  This module automatically adds journal entries generated from the
+*Statement Operation Templates* if a rule matches with the current
+situation, so there is nothing special to do once the rules are
+configured.
+
```
Comment (pedrobaeza) [inline account_statement_operation_rule/README.rst]: Please add new "Try me on runbot" button

File: account_statement_operation_rule/__openerp__.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{'name': 'Bank Statement Operation Rules',
+ 'version': '1.0',
```
Comment (pedrobaeza) [inline account_statement_operation_rule/__openerp__.py]: Use new version numbering scheme: 8.0.1.0.0

File: account_statement_operation_rule/README.rst (hunk 8 above)
Comment (guewen) [inline account_statement_operation_rule/README.rst:18]: This is a definition list, not a bullet points list (http://docutils.sourceforge.net/docs/user/rst/quickref.html#definition-lists).

**Comment by guewen** [comment]
@pedrobaeza done

**Comment by pedrobaeza** [comment]
Why runbot is failing asking for the module "connector"?

**Comment by pedrobaeza** [comment]
OK, it seems to be due to the account-financial-tools dependency. I have forced the rebuild.

**Comment by pedrobaeza** [comment]
Anyways, seems :+1:

**Comment by fclementic2c** [comment]
:+1: 
Functional test ok

**Comment by guewen** [comment]
I rebased

**Comment by vrenaville** [comment]
:+1:
